src/app/pages/ParaTimesPage/TransactionAmount/index.tsx

```diff
@@ -0,0 +1,231 @@
+import React, { useEffect } from 'react'
+import { useDispatch } from 'react-redux'
+import { Box, Button, Form, FormField, Text, TextInput, Tip } from 'grommet'
+import { Trans, useTranslation } from 'react-i18next'
+import BigNumber from 'bignumber.js'
+import { formatBaseUnitsAsRose, isAmountGreaterThan } from 'app/lib/helpers'
+import { paraTimesActions } from 'app/state/paratimes'
+import { AlertBox } from 'app/components/AlertBox'
+import { defaultWithdrawFeeAmount } from 'app/lib/transaction'
+import { StringifiedBigInt } from 'types/StringifiedBigInt'
+import { AmountFormatter } from '../../../components/AmountFormatter'
+import { ParaTimeContent } from '../ParaTimeContent'
+import { ParaTimeFormFooter } from '../ParaTimeFormFooter'
+import { useParaTimes } from '../useParaTimes'
+import { useParaTimesNavigation } from '../useParaTimesNavigation'
+import { FeesSection } from './FeesSection'
+
+function getMaxAmountValue(
+  isDepositing: boolean,
+  accountBalance: StringifiedBigInt,
+  feeAmount: string,
+  balanceInBaseUnit: boolean,
+  consensusDecimals: number,
+  runtimeDecimals: number,
+): StringifiedBigInt {
+  const balance = balanceInBaseUnit
+    ? accountBalance
+    : new BigNumber(accountBalance).shiftedBy(consensusDecimals - runtimeDecimals).toFixed(0)
+  const maxValue =
+    isDepositing && !feeAmount
+      ? accountBalance
+      : BigInt(balance) - BigInt(feeAmount || defaultWithdrawFeeAmount)
+  return maxValue.toString()
+}
```

Comment:
I don't understand any of this code.

useParaTimes starts with `balance`, `accountBalance`, and `walletBalance` (depends on isDepositing). Then it renames `walletBalance` to `balance`.

TransactionAmount takes useParaTimes's `balance` and getMaxAmountValue renames it to `accountBalance`. Then getMaxAmountValue computes a `balance` and `maxValue` (depends on isDepositing).

And `balanceInBaseUnit`, `decimals`, and `isWalletEmpty` refer to only one of those balances.. probably. I also no longer like `isWalletEmpty` name, since it sounds like there are no accounts in a wallet.

And getMaxAmountValue's `runtimeDecimals` comes from useParaTimes's `decimals` - which can be consensusDecimals or runtimeDecimals